docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+&nbsp;&nbsp;&nbsp;3.5 Executor Layer\
+&nbsp;&nbsp;&nbsp;3.6 Storage Layer\
+**4. Implementation**\
+&nbsp;&nbsp;&nbsp;4.1 Module Planner Feature\
+&nbsp;&nbsp;&nbsp;4.2 Checker Feature\
+&nbsp;&nbsp;&nbsp;4.3 Cap Caculator Feature\
+&nbsp;&nbsp;&nbsp;4.4 Reminder Feature\
+&nbsp;&nbsp;&nbsp;4.5 Postpone Feature\
+**5. Appendix A Product Scope**\
+**6. Appendix B User Stories** \
+**7. Appendix C Use Cases** \
+**8. Appendix D Non-funcitonal Requirements** \
+**9. Appendix E Glossary** \
+**10. Appendix F. Instruction for Manual Testing**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Setting Up
 
+1.  Ensure that you have Java 11 or above installed.
+2.  Download the latest version of  `Duke`  from  [Our Release Page](https://github.com/AY2021S1-CS2113-T13-2/tp/releases/tag/v1.0).
+3.  Copy the file to the folder you want to use as the home folder for your Mobile Nusmod.
+4.  Open the Command Prompt if you are running on Windows or Terminal if you are running on Mac OS.
+5.  Navigate to your home folder and type ‘java -jar domnus.jar’
+6.  Type ‘bye’ to terminate your session.
 
-## Product scope
+## 3. Design
+
+### 3.1 Architecture
+The **Architecture Diagram** below represents a high-level design overview of the App. Specifically, it is done with an **N-tier architectural style**, where the higher layers make use of services provided by lower layers. 
+
+![here](Images/Architecture_Diagram.PNG)
+
+
+**3.2 Main Layer**<br>
+For the `main` layer, it contains a single class known as `Duke`. 
+
+**3.3 UI Layer**<br>
+Main gets user input and displays messages through the use of UI component. 
+The UI layer entails the package *visualize*, which contains classes *ColoredString*, *Bitmap*, *UI*, *Cli*, 
+*FancyCli* and enumerations *Color* and *Sprite* in the following structure:
+![uml](Images/Package%20visualize.png)
+
+UI's interaction with user<br>
+UI gets user input through *nextline()*, and renders strings as a user-comprehensible interface through 
+*update(String, Data)*.
+
+UI's interaction with the rest of the program<br>
+UI passes the user's input string out to the Duke object, which then passes the string to the Command Interpreter layer.
+UI also reads data from the Data object for refreshing purposes, but does not modify it. 
+
+**1.3 Command Interpreter Layer**<br>
+Upon receiving command from the UI, Duke would pass the entire user input into Command Interpreter (CI)
+
+**1.4 Execute Layer**<br>
+Once CI processed the user input, duke proceeds to redirect the input to Execute for execution of action. 
+
+**1.5 Storage Layer**<br>
+Once CI processed the user input, duke proceeds to redirect the input to Execute for execution of action. 
+
+**1.6 Flow of DOMSUN**<br>
+The sequence diagram below shows the main interaction of classes with each other throughout the whole lifecycle of DOMSUM.
+![uml](Images/DOMSUM_Main_Flow.png)
```

Comment:
This diagram have some indentation before it, which is not consistent with the other ones. And I found it's because there is a line "lifecycle of DOMSUM." before the diagram. So please try separate them.
![image](https://user-images.githubusercontent.com/59434361/97666908-99690280-1ab9-11eb-8c0e-062263d28e79.png)